R/importM.COSMIC.R

```diff
@@ -0,0 +1,250 @@
+#' mutation barcode is not available for COSMIC data, and that the sampleID
+#' for this data is specific to COSMIC data)), row names are HGNC gene
+#' symbols, and the values represent copy number.
+#'
+#' @section COSMIC CNA :
+#' The format of the COSMIC CNA data is as follows.
+#' (through sftp, run get with this path:
+#' /files/grch38/cosmic/v80/CosmicCompleteCNA.tsv.gz)
+#'
+#' [col number] Heading
+#' [1] CNA_ID
+#' [2] Gene ID
+#' [3] Gene Name
+#' [4] Sample ID
+#' [5] Tumour ID
+#' [6] Primary Site
+#' [7-9] Site Subtype 1-3
+#' [10] Primary Histology
+#' [11-13] Histology Subtype 1-3
+#' [14] Sample Name
+#' [15] Total Copy Number
+#' [16] Minor Allele CN
+#' [17] Mutation Type
+#' [18] Study ID
+#' [19] GRCh Coordinate System
+#' [20] Genomic Coordinates of Variation
+#'
+#' @param fNameCNA The path to a TSV file with CNA data from COSMIC.
+#' @param outFName The path to an output file for the rCNA RDS object. Only
+#'   required if writeToFile == TRUE. Must be .rds file type.
+#' @param writeToFile Controls whether to serialize rCNA RDS object.
+#'   FALSE by default.
+#' @param silent Controls whether output to console should be suppressed. FALSE
+#'   by default.
+#' @param writeLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @return Return rCNA object of COSMIC CNA data as RDS.
+#'
+#' @family importM.COSMIC, importM.TCGA, importM.GISTIC2, importM.ProjectGenie
+#'
+#'   ## @seealso \code{\link{fastMap}} fastMap() is used internally to
+#'   map COSMIC IDs to gene symbols.
+#'
+#' @examples
+#' \dontrun{
+#' fNameCNA <- "./../test/COSMIC/testCosmicCNA.tsv"
+#'
+#' importM.COSMIC(fNameCNA,
+#'                writeToFile = FALSE,
+#'                silent = TRUE,
+#'                writeLog = FALSE)
+#' }
+#' @export
+importM.COSMIC <- function(fNameCNA,
+                           outFName,
+                           writeToFile = FALSE,
+                           silent = FALSE,
+                           writeLog = TRUE) {
+
+    # ==== PARAMETER TYPE VALIDATION ==========================================
+
+    # General parameter checks
+    cR <- character()
+    cR <- c(cR, .checkArgs(fNameCNA,     like = "FILE_E",   checkSize = TRUE))
+    cR <- c(cR, .checkArgs(writeToFile,  like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(silent,       like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(writeLog,     like = logical(1), checkSize = TRUE))
+
+    if(length(cR) > 0) {
+        stop(cR)
+    }
+
+    if (writeToFile) {
+        # Make sure output file name is provided if writeToFile is TRUE.
+        if (missing(outFName)) {
+            stop("No output file specified in outFName.")
+        }
+
+        # Check that outFName is a string.
+        if (!is.character(outFName)) {
+            stop("outFName is not a valid string.")
+        }
+
+        # Check that outFName is of type .rds
```

Comment:
I don't think you have to enforce that. It makes sense - but there's no real reason we have to insist on it.